ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java

```diff
@@ -37,44 +38,60 @@
   protected long sentBytes = 0;
   protected final BlobProperties properties;
   protected final BlobType blobType;
+  protected ByteBuffer materializedBlobBuffer;
```

Comment:
minor: `materializedBlob` or `blob` should do I think (esp since `usermetadata` is not named `usermetadataBuffer`